tools/dotcms-cli/api-data-model/src/main/java/com/dotcms/api/traversal/TreeNode.java

```diff
@@ -119,7 +137,7 @@ public TreeNode cloneAndFilterAssets(final boolean live, final String language,
         boolean includeAssets = includeAssets();
         if (includeAssets && this.assets != null) {
             List<AssetView> filteredAssets = this.assets.stream()
-                    .filter((asset) -> {
+                    .filter(asset -> {
```

Comment:
This method could do with decomposing a bit.   An example from ChatGPT
```

public TreeNode cloneAndFilterAssets(final boolean live, final String language,
                                         final boolean showEmptyFolders,
                                         final boolean filterForPushChanges) {

    TreeNode newNode = new TreeNode(this.folder, true);

    filterAndAddAssetsToNode(newNode, live, language);
    cloneAndFilterChildren(newNode, live, language, showEmptyFolders, filterForPushChanges);

    return newNode;
}

private void filterAndAddAssetsToNode(TreeNode newNode, boolean live, String language) {
    Optional.ofNullable(this.assets)
            .ifPresent(assets -> {
                List<AssetView> filteredAssets = assets.stream()
                        .filter(asset -> assetStatusMatches(asset, live) && asset.lang().equalsIgnoreCase(language))
                        .collect(Collectors.toList());
                newNode.assets(filteredAssets);
            });
}

private boolean assetStatusMatches(AssetView asset, boolean live) {
    return live ? asset.live() : asset.working();
}

private void cloneAndFilterChildren(TreeNode newNode, boolean live, String language, boolean showEmptyFolders,
                                    boolean filterForPushChanges) {

    this.children.stream()
            .filter(child -> !child.folder().explicitGlobExclude())
            .map(child -> child.cloneAndFilterAssets(live, language, showEmptyFolders, filterForPushChanges))
            .filter(clonedChild -> shouldAddChild(clonedChild, showEmptyFolders, filterForPushChanges))
            .forEach(newNode::addChild);
}

private boolean shouldAddChild(TreeNode clonedChild, boolean showEmptyFolders, boolean filterForPushChanges) {
    return showEmptyFolders || !clonedChild.assets.isEmpty() ||
           (filterForPushChanges ? meetsPushChangeCriteria(clonedChild) : hasAssetsInSubtree(clonedChild)) &&
           (clonedChild.folder.implicitGlobInclude() || hasIncludeInSubtree(clonedChild));
}

private boolean meetsPushChangeCriteria(TreeNode clonedChild) {
    return clonedChild.folder().markForPush().isPresent() || clonedChild.folder().markForDelete().isPresent() ||
           hasAssetsWithChangesInSubtree(clonedChild) || hasAFolderWithChangesInSubtree(clonedChild);
}

```

Reply:
Again, I'm just fixing a bug, But I agree this class, TreeNode shouldn't hold that much logic in it. and the use of Optional there is anti-pattern